parlai/agents/rag/retrieve_api.py

```diff
@@ -0,0 +1,132 @@
+#!/usr/bin/env python3
+# Copyright (c) Facebook, Inc. and its affiliates.
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+"""
+APIs for retrieving a list of "Contents" using an "Search Query".
+
+The term "Search Query" here refers to any abstract form of input string. The definition
+of "Contents" is also loose and depends on the API.
+"""
+
+from abc import ABC, abstractmethod
+import requests
+from typing import Any, Dict, List
+
+from parlai.core.opt import Opt
+from parlai.utils import logging
+
+
+CONTENT = 'content'
+DEFAULT_NUM_TO_RETRIEVE = 5
+
+
+class RetrieverAPI(ABC):
+    """
+    Provides the common interfaces for retrievers.
+
+    Every retriever in this modules must implement the `retriev` method.
+    """
+
+    def __init__(self, opt: Opt):
+        self.skip_query_token = opt['skip_retrieval_token']
```

Comment:
I thought because this is supposed to be an abstract class (used for any retrieval) maybe that token depends on the class implementation. Maybe I could put the constants in `SearchEngineRetriever`?